File: codeforces/488A/Giga.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main() {
	int floor;
	cin >> floor;

	int count = 0;
	bool stop = false;
	while(!stop) {
		string floorStr = to_string(floor);
		if(floorStr.find("8") != string::npos) {
			cout << count << "\n";
			stop = true;
		}
		else {
			count++;
			floor++;
		}
	}
}

File: codeforces/488A/Giga.java
import java.util.Scanner;
import java.lang.Integer; 

public class Giga {
	public static void main(String [] args) {
		Scanner input = new Scanner(System.in);
		int floor = input.nextInt() + 1;
		boolean stop = false;
		int count = 1;

		while(!stop) {
			String floorStr = Integer.toString(floor);
			if(floorStr.contains("8")) {
				System.out.println(count);
				stop = true;
			}
			else {
				floor++;
				count++;
			}
		}
	}
}